filebeat/module/apache/access/config/access.yml

```diff
@@ -8,4 +8,4 @@ processors:
   - add_fields:
       target: ''
       fields:
-        ecs.version: 1.6.0
+        ecs.version: 1.7.0
```

Comment:
I'm wondering if we should wait to bump up the ecs version in each of the beats/modules until we have evaluated them.  I've taken this value to mean that we have looked at the module and to the best of our ability we are mapping values to the available ECS fields in this version.  So if Apache says it is ECS 1.7.0, then I might expect it to have `url.subdomain` (new ECS 1.7 field) mapped, where if it was reporting ECS 1.6.0 I would know that it wouldn't have that field.  
Thoughts?

Reply:
So, generally I'd agree with that, in this case though one of the large changes for 1.7.0 experimental is that we get the remapping of a bunch of fields from `keyword` to `wildcard` -- in this case the fact that that's such a major mapping change makes me think we should just update all to 1.7.0 initially and just track the additional field enhancements in an issue. At that point we're using this to signal "yeah, we're shipping 1.7.0 schema" even though we don't have all final fields filled in yet before the release. The alternative, saying "we're shipping 1.6.0, but these fields are actually 1.7.0 experimental wildcards" seems less ideal to me.

Reply:
darn wildcards 😄 

that makes sense.

Reply:
darn grey areas 😂 

One thing to consider is that the `ecs.version` can either be a statement about:

* which fields are populated in the event
* the mapping itself. Like whether it uses `keyword` or `wildcard`

I've always taken it to mean the former. Although I'm thinking users may want to start doing queries that depend on the presence of `wildcard`, and want to exclude older indices. Using `ecs.version` for that would probably be the easiest way.

On the "shape of the event" aspect, we should consider that ECS 1.7 introduced the following fields:

* Mainline ECS: `http.[request|response].mime_type` and `threat.technique.subtechnique`
* Experimental: `user.target.*`, `user.effective.*` and `user.changes.*`


In general, I would favor the conservative approach of only updating that number when the source has been adjusted to populate new relevant fields (also ok to upgrade if the source is unaffected).

But in this case, I would err on the side of updating the version to 1.7.0 aggressively, so that users can use `ecs.version` to guard potentially expensive queries from hitting older `keyword`-based indices.

Reply:
@webmat I like that general guidance. I'll see about documenting that somewhere in the repo in developer docs in a subsequent PR. If this weren't quite so large of a change, definitely agree that it would make sense just to update as we added the fields.